### PR TITLE
docs(GlobalHeader): Polish GlobalHeader / AppHeader docs

### PR DESCRIPTION
**Pull request by katiezutter**
### Overview

Touch up Storybook documentation for all GlobalHeader and AppHeader components.

**No code changes.**

Please look over the Storybook docs for all header components in Labs > Experimental > Molecules. Would love to hear any suggestions for improvement because I know I have blindspots, having worked on these components for awhile now!

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [EGG-660](https://codecademy.atlassian.net/browse/EGG-660)
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@29.1.2-alpha.5e8230.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6033c823963d4c40af48d69e--gamut-preview.netlify.app>https://6033c823963d4c40af48d69e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6033c823963d4c40af48d69e>Deploy Logs</a></p>

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeader.stories.mdx [hunk 1]
```diff
@@ -55,9 +60,9 @@ import { MOCK_DROPDOWN_1 } from './MockAppHeaderElements';
...
 
-AppHeader is not intended to be used alone, but rather by GlobalHeader which determines the composition of each of the Pro, Free, and Anon versions of Codecademy's global navigation header and passes the corresponding items to AppHeader.
+AppHeader is not intended to be used directly by the consumer, but rather by <LinkTo kind="Labs/Experimental/Molecules/GlobalHeader" story="GlobalHeader">GlobalHeader</LinkTo> which determines the composition of each of the versions of Codecademy's global navigation header and passes the corresponding items to this component.
```
Comment (saghdaey) [inline packages/styleguide/stories/Labs/Experimental/Molecules/AppHeader.stories.mdx:65]: nit: determines the composition of each version of Codecademy's  global nav

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderDropdown.stories.mdx [hunk 2]
```diff
@@ -10,7 +11,13 @@ import { MOCK_PROFILE_DROPDOWN } from './MockAppHeaderElements';
 <Meta
   title={title}
   component={AppHeaderDropdown}
-  parameters={{ layout: 'padded', status: 'volatile', source: 'gamut-labs' }}
+  parameters={{
+    layout: 'padded',
+    status: 'volatile',
+    source: 'gamut-labs',
+    subtitle:
+      'An AppHeader subcomponent, used for both link dropdowns and the profile dropdown.',
```
Comment (saghdaey) [inline packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderDropdown.stories.mdx:19]: "link dropdowns" is a little confusing; the profile dropdown is also just a drop down of links. Maybe we should say used for basic header drop downs and the profile dropdown?

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderSubMenuMobile.stories.mdx [hunk 3]
```diff
@@ -11,23 +12,28 @@ import {
...
   }}
 />
 
-The AppHeaderSubMenuMobile component is used in the mobile version of the AppHeader.
-Similar to in the desktop App Header, when an AppHeaderItem has type "dropdown", clicking on it opens a dropdown menu, on smaller screens, clicking on an AppHeaderItem of type "dropdown" opens the AppHeaderSubMenuMobile component. The component contains a title indicating the category of the menu (ex resources, community) and a list of links (forum, codecademy blog, etc).
+This component is used in the mobile version of the header (<LinkTo kind="Labs/Experimental/Molecules/AppHeaderMobile" story="AppHeaderMobile">AppHeaderMobile</LinkTo>).
 
-## SubMenu for mobile view
+Similar to in the desktop <LinkTo kind="Labs/Experimental/Molecules/AppHeader" story="AppHeader">AppHeader</LinkTo>, when an AppHeaderItem has type "dropdown", clicking on it opens a dropdown menu, on smaller screens, clicking on an AppHeaderItem of type "dropdown" opens the <LinkTo kind="Labs/Experimental/Molecules/AppHeaderSubMenuMobile" story="AppHeaderSubMenuMobile">AppHeaderSubMenuMobile</LinkTo> component. The component contains a title indicating the category of the menu and a list of links.
```
Comment (saghdaey) [inline packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderSubMenuMobile.stories.mdx:27]: nit: use a semicolon after the desktop 

Similar to in the desktop AppHeader, when an AppHeaderItem has type "dropdown", clicking on it opens a dropdown menu**;** (semicolon) on smaller screens, clicking on an AppHeaderItem of type "dropdown" opens the AppHeaderSubMenuMobile component.

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx [hunk 4]
```diff
@@ -21,46 +22,68 @@ import { PropsTable } from '~styleguide/blocks';
...
+GlobalHeader determines whether to render the desktop or mobile version of the global header (<LinkTo kind="Labs/Experimental/Molecules/AppHeader" story="AppHeader">AppHeader</LinkTo> or <LinkTo kind="Labs/Experimental/Molecules/AppHeaderMobile" story="AppHeaderMobile">AppHeaderMobile</LinkTo>, respectively) based on the viewport size.
+
+GlobalHeader takes in several props. `type` determines which header to render based on the user role (anonymous, free, or pro), or a loading state. `type` will also determine which of Search and/or Notification components need to be passed in: Search will appear for all users, whereas Notifications will only appear for logged-in users. The consumer must pass components for both the desktop and mobile header for either of these render props. `action` is a function which will be defined by the consumer and handles any routing/tracking that needs to be done. `variant` is a prop which may only be used when `type="anon"`, and provides variations of the anonymous header based on the user's location: there are different headers for the landing page, login page, and signup page.
```
Comment (saghdaey) [inline packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:46]: nice

File: packages/styleguide/stories/Labs/Experimental/Molecules/MockAppHeaderElements.tsx [hunk 5]
```diff
@@ -70,7 +70,7 @@ export const MOCK_DROPDOWN_2: AppHeaderSimpleDropdownItem = {
 export const MOCK_PROFILE_DROPDOWN: AppHeaderProfileDropdownItem = {
   avatar:
     'https://www.gravatar.com/avatar/1c959a9a1e2f9f9f1ac06b05cccc1d60?s=150&d=retro',
-  userDisplayName: 'Katie',
+  userDisplayName: 'Codey',
```
Comment (saghdaey) [inline packages/styleguide/stories/Labs/Experimental/Molecules/MockAppHeaderElements.tsx:73]: boo

**Comment by saghdaey** [comment]
Looks good to me!! couple of v small things.
I would suggest requesting another Egg to review as well since I have a lot f knowledge of these components + practice using them!

File: packages/styleguide/stories/Labs/Experimental/Molecules/MockAppHeaderElements.tsx (hunk 5 above)
Comment (katiezutter) [reply packages/styleguide/stories/Labs/Experimental/Molecules/MockAppHeaderElements.tsx:73]: loll 😅

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@29.1.2-alpha.ad07d0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6033d27e24700c3efd488d37--gamut-preview.netlify.app>https://6033d27e24700c3efd488d37--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6033d27e24700c3efd488d37>Deploy Logs</a></p>

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeader.stories.mdx [hunk 6]
```diff
@@ -10,7 +11,11 @@ import { MOCK_DROPDOWN_1 } from './MockAppHeaderElements';
 <Meta
   title="Labs/Experimental/Molecules/AppHeader"
   component={AppHeader}
-  parameters={{ layout: 'padded', status: 'volatile' }}
+  parameters={{
+    layout: 'padded',
+    status: 'volatile',
+    subtitle: `A flexible header component, used by GlobalHeader. Takes in an object with AppHeaderItems and maps those items to their corresponding AppHeader components.`,
```
Comment (sylvanasantos) [inline packages/styleguide/stories/Labs/Experimental/Molecules/AppHeader.stories.mdx:17]: super nit and also just a question for my own knowledge: is it necessary to state that it used by the global header? wondering if there was a specific reason for adding that detail

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeader.stories.mdx [hunk 7]
```diff
@@ -55,9 +60,9 @@ import { MOCK_DROPDOWN_1 } from './MockAppHeaderElements';
   }}
 />
 
-AppHeader component for use in GlobalHeader. Uses AppBar to arrange items to the left or right side of the navigation bar. These items are passed in as objects of type AppHeaderItem, and map to AppHeaderLogo, AppHeaderLink, AppHeaderDropdown, TextButton, FillButton, or a custom component passed in by the consumer.
+Uses <LinkTo kind='Molecules/AppBar' story="Header">AppBar</LinkTo> to arrange items determined by <LinkTo kind="Labs/Experimental/Molecules/GlobalHeader" story="GlobalHeader">GlobalHeader</LinkTo> to the left or right side of the navigation bar. These items are passed in as objects of type AppHeaderItem, and map to <LinkTo kind="Labs/Experimental/Molecules/AppHeaderLogo" story="AppHeaderLogo">AppHeaderLogo</LinkTo>, <LinkTo kind="Labs/Experimental/Molecules/AppHeaderLink" story="AppHeaderLink">AppHeaderLink</LinkTo>, <LinkTo kind="Labs/Experimental/Molecules/AppHeaderDropdown" story="AppHeaderDropdown">AppHeaderDropdown</LinkTo>, <LinkTo kind="Atoms/Button" story="TextButton">TextButton</LinkTo>, <LinkTo kind="Atoms/Button" story="FillButton">FillButton</LinkTo>, or a custom component passed in by the consumer.
```
Comment (sylvanasantos) [inline packages/styleguide/stories/Labs/Experimental/Molecules/AppHeader.stories.mdx:63]: hmm, similar to comment above. is there a need to say that the items are determined by global header? couldn't a client potentially use this component without the global header component? or are you trying to encourage people to not use this on its own?

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeader.stories.mdx [hunk 8]
```diff
@@ -55,9 +60,9 @@ import { MOCK_DROPDOWN_1 } from './MockAppHeaderElements';
...
 
-AppHeader is not intended to be used alone, but rather by GlobalHeader which determines the composition of each of the Pro, Free, and Anon versions of Codecademy's global navigation header and passes the corresponding items to AppHeader.
+AppHeader is not intended to be used directly by the consumer, but rather by <LinkTo kind="Labs/Experimental/Molecules/GlobalHeader" story="GlobalHeader">GlobalHeader</LinkTo> which determines the composition of each version of Codecademy's global navigation header and passes the corresponding items to this component.
```
Comment (sylvanasantos) [inline packages/styleguide/stories/Labs/Experimental/Molecules/AppHeader.stories.mdx:65]: ignore everything i just said

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderSubMenuMobile.stories.mdx [hunk 9]
```diff
@@ -11,23 +12,28 @@ import {
...
   }}
 />
 
-The AppHeaderSubMenuMobile component is used in the mobile version of the AppHeader.
-Similar to in the desktop App Header, when an AppHeaderItem has type "dropdown", clicking on it opens a dropdown menu, on smaller screens, clicking on an AppHeaderItem of type "dropdown" opens the AppHeaderSubMenuMobile component. The component contains a title indicating the category of the menu (ex resources, community) and a list of links (forum, codecademy blog, etc).
+This component is used in the mobile version of the header (<LinkTo kind="Labs/Experimental/Molecules/AppHeaderMobile" story="AppHeaderMobile">AppHeaderMobile</LinkTo>).
 
-## SubMenu for mobile view
+Similar to in the desktop <LinkTo kind="Labs/Experimental/Molecules/AppHeader" story="AppHeader">AppHeader</LinkTo>, when an AppHeaderItem has type "dropdown", clicking on it opens a dropdown menu; on smaller screens, clicking on an AppHeaderItem of type "dropdown" opens the <LinkTo kind="Labs/Experimental/Molecules/AppHeaderSubMenuMobile" story="AppHeaderSubMenuMobile">AppHeaderSubMenuMobile</LinkTo> component. The component contains a title indicating the category of the menu and a list of links.
```
Comment (sylvanasantos) [inline packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderSubMenuMobile.stories.mdx:27]: typo here "similar to in" :)

File: packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx [hunk 10]
```diff
@@ -21,46 +22,68 @@ import { PropsTable } from '~styleguide/blocks';
...
+header content using <LinkTo kind='Molecules/AppBar' story="Header">AppBar</LinkTo> (with a `left` and `right` `AppBarSection`). The <LinkTo kind="Labs/Experimental/Molecules/AppHeader" story="AppHeader">AppHeader</LinkTo>
+has no knowledge of the user; it just takes as a prop an object that has a left property
+(an array of `AppHeaderItem`) to render on the left side of the header and a right
+property (an array of `AppHeaderItem`) to render on the right side of the header.
+These item objects which are passed in are mapped to <LinkTo kind="Labs/Experimental/Molecules/AppHeader" story="AppHeader">AppHeader</LinkTo> components. Things work in a similar way for the mobile menu, except there is an additional key, `mainMenu` passed into <LinkTo kind="Labs/Experimental/Molecules/AppHeaderMobile" story="AppHeaderMobile">AppHeaderMobile</LinkTo> which determines which elements are rendered within the main mobile menu and the mobile submenus.
```
Comment (sylvanasantos) [inline packages/styleguide/stories/Labs/Experimental/Molecules/GlobalHeader.stories.mdx:57]: amazing!

**Review by sylvanasantos — APPROVED** [review]
had a pointless dialogue with myself while reviewing. one small typo, otherwise looks fantastic!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@29.1.3-alpha.5f8d7e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6033e4e15ded7d00d529a98a--gamut-preview.netlify.app>https://6033e4e15ded7d00d529a98a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6033e4e15ded7d00d529a98a>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
the documentation is 🔥  i'm seeing some small inconsistencies in spacing/styling but i'm almost certain this is gamut-specific :

image

**Comment by katiezutter** [comment]
@dreamwasp yeah, I think that spacing is a storybook thing 😊 if you go to the Canvas tab (vs the Docs tab), you'll see a better representation of the components!

**Review by JoshuaKGoldberg — APPROVED** [review]
🌮 !

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@29.3.1-alpha.522102.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6033f3e321875621760460c7--gamut-preview.netlify.app>https://6033f3e321875621760460c7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6033f3e321875621760460c7>Deploy Logs</a></p>

**Pull request merged**